libraries/ESP8266WebServer/src/ESP8266WebServer-impl.h

```diff
@@ -296,12 +296,30 @@ void ESP8266WebServerTemplate<ServerType>::handleClient() {
   bool keepCurrentClient = false;
   bool callYield = false;
 
-  DBGWS("http-server loop: conn=%d avail=%d status=%s\n",
-    _currentClient.connected(), _currentClient.available(),
-    _currentStatus==HC_NONE?"none":
-    _currentStatus==HC_WAIT_READ?"wait-read":
-    _currentStatus==HC_WAIT_CLOSE?"wait-close":
-    "??");
+#ifdef DEBUG_ESP_HTTP_SERVER
+
+  struct compare_s
+  {
+    uint8_t connected;
+    int available;
+    HTTPClientStatus status;
+    bool operator != (const compare_s& o) { return memcmp(this, &o, sizeof(o)) != 0; }
```

Comment:
Padding? Just write every member instead of comparing raw mem?
*(or someone copies this into their virtual class accidentally :>)*

Reply:
I wonder why c++ does not define a default operator by implicitly calling it on every members.

Reply:
https://en.cppreference.com/w/cpp/language/default_comparisons (c++20)
https://open-std.org/JTC1/SC22/WG21/docs/papers/2017/p0515r3.pdf